src/ImageSharp/Formats/Jpeg/Common/Decoder/JpegBlockPostProcessor.cs

```diff
@@ -11,145 +11,80 @@ namespace SixLabors.ImageSharp.Formats.Jpeg.Common.Decoder
         }
 
+        /// <summary>
+        /// Processes 'sourceBlock' producing Jpeg color channel values from spectral values:
+        /// - Dequantize
+        /// - Applying IDCT
+        /// - Level shift by +128, clip to [0, 255]
+        /// - Copy the resultin color values into 'destArea' scaling up the block by amount defined in <see cref="subSamplingDivisors"/>
+        /// </summary>
         public void ProcessBlockColorsInto(
-            IRawJpegData decoder,
-            IJpegComponent component,
             ref Block8x8 sourceBlock,
             BufferArea<float> destArea)
         {
-            this.QuantizeAndTransform(decoder, component, ref sourceBlock);
+            ref Block8x8F b = ref this.SourceBlock;
+            sourceBlock.CopyToFloatBlock(ref b);
+
+            // Dequantize:
+            b.MultiplyInplace(ref this.DequantiazationTable);
 
-            this.data.WorkspaceBlock1.NormalizeColorsInplace();
+            FastFloatingPointDCT.TransformIDCT(ref b, ref this.WorkspaceBlock1, ref this.WorkspaceBlock2);
 
             // To conform better to libjpeg we actually NEED TO loose precision here.
             // This is because they store blocks as Int16 between all the operations.
-            // Unfortunately, we need to emulate this to be "more accurate" :(
-            this.data.WorkspaceBlock1.RoundInplace();
-
-            Size divs = component.SubSamplingDivisors;
-            this.data.WorkspaceBlock1.CopyTo(destArea, divs.Width, divs.Height);
-        }
-
-        /// <summary>
-        /// Holds the "large" data blocks needed for computations.
-        /// </summary>
-        [StructLayout(LayoutKind.Sequential)]
-        public struct ComputationData
-        {
-            /// <summary>
-            /// Source block
-            /// </summary>
-            public Block8x8F SourceBlock;
-
-            /// <summary>
-            /// Temporal block 1 to store intermediate and/or final computation results
-            /// </summary>
-            public Block8x8F WorkspaceBlock1;
-
-            /// <summary>
-            /// Temporal block 2 to store intermediate and/or final computation results
-            /// </summary>
-            public Block8x8F WorkspaceBlock2;
-
-            /// <summary>
-            /// The quantization table as <see cref="Block8x8F"/>
-            /// </summary>
-            public Block8x8F QuantiazationTable;
-
-            /// <summary>
-            /// The jpeg unzig data
-            /// </summary>
-            public UnzigData Unzig;
-
-            /// <summary>
-            /// Creates and initializes a new <see cref="ComputationData"/> instance
-            /// </summary>
-            /// <returns>The <see cref="ComputationData"/></returns>
-            public static ComputationData Create()
+            // To be "more accurate", we need to emulate this by rounding!
+            if (SimdUtils.IsAvx2CompatibleArchitecture)
```

Comment:
for readability shouldn't this be a single call to  `this.WorkspaceBlock1.NormalizeColorsAndRoundInplace();` with the if check inside the callled method.

Reply:
Agree, will do it.